### ISSUE TITLE
Uberti Cattleman issues with nested containers

### ISSUE THREAD

**Issue by maibock-ellie** [CleverRaven/Cataclysm-DDA#44090]
# Bug Description

 Uberti Cattleman can not eject casings as expected due to the new nested containers. Casings are gone entirely.

# Steps to Reproduce

1. Obtain an Uberti Cattleman
2. (R)eload with .45 colt
3. (F)ire the six rounds
4. (U)nload it
5. Ta-da, casings disappear from existence.

# Expected Behavior

When (U)nloaded, casings should either drop to the ground, or end up in the inventory.

# Screenshots

No screenshots available at the moment.

# Versions and configuration

OS: Windows 10
Game Version: 0.E-5833-g13fd30e
Graphics Version: Tiles
Ingame Language: English
Mods Loaded: dda

**Comment by wapcaplet** [comment]
Confirmed, though the problem appears to occur when the weapon is fired, rather than when it is reloaded.

Upon firing a single round from an Uberti Cattleman, I get this message:

```
 DEBUG    : tried to put an item (45colt_casing) count (1) in a container (colt_saa) that cannot contain it: pocket with type (CONTAINER) not found

 FUNCTION : ret_val<bool> item::put_in(const item &, item_pocket::pocket_type)
 FILE     : src/item.cpp
 LINE     : 1103
```

The same kind of error happens with another revolver, the S&W Model 10:

```
 DEBUG    : tried to put an item (38_casing) count (1) in a container (model_10_revolver) that cannot contain it: pocket with type (CONTAINER) not found
```

What this error means is the "pocket" in the gun that holds the ammo cannot hold casings. Generic "CONTAINER" type pockets can hold arbitrary items, but this pocket is a "MAGAZINE" designed only to hold ammo:

https://github.com/CleverRaven/Cataclysm-DDA/blob/a252c8392e50401b576eb85c33809e4b56aee26c/data/json/items/gun/45colt.json#L85

I thought perhaps adding the `RELOAD_EJECT` flag to the gun would allow it to hold onto its casings until reload time, but that did not work for me. It looks like we may need to make some code adjustments for revolvers, so they can continue to hold ammo casings until they are reloaded.

**Comment by maibock-ellie** [comment]
Ah! That's what I meant to say. Sorry for not providing much information, as I quickly typed this up before heading off to bed last night. Like you said, there will probably have to be some code adjustments for all revolvers. I'll give it a look later when I have some actual free time and see if I can make said adjustments in a PR.

**Comment by OtpalTS** [comment]
This seems to affect brass catchers as well.

**Comment by DoctorVanGogh** [comment]
> This seems to affect brass catchers as well.

Can confirm that this seems to be a *generalized* brass catcher issue (either built-in or attached).
Any rifle with a brass catcher attached will error `tried to put an item (...) count (...) in a container (...)
  that cannot contain it` on shot.

**Comment by wapcaplet** [comment]
Problems with brass catchers (#44065) should be fixed by #42455
